### PR TITLE
Removes letkf task from gdasapp

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
LETKF task scripts moved to https://github.com/NOAA-EMC/global-workflow/pull/2564

Also changed `PYTHONPATH` in ctests to know where to find the future home of Python scripts in global-workflow.

Partially addresses https://github.com/NOAA-EMC/GDASApp/issues/1091

### CODE REVIEW

**Review by apchoiCMD — APPROVED** [review]

File: test/soca/gw/CMakeLists.txt [hunk 1]
```diff
@@ -77,7 +77,7 @@ foreach(jjob ${jjob_list})
     WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/test/soca/gw/testrun)
   set_tests_properties(${test_name}
     PROPERTIES
-    ENVIRONMENT "PYTHONPATH=${PROJECT_SOURCE_DIR}/ush/ioda/bufr2ioda::${PROJECT_SOURCE_DIR}/ush:${PROJECT_BINARY_DIR}/../lib/python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}:$ENV{PYTHONPATH}")
+    ENVIRONMENT "PYTHONPATH=${PROJECT_SOURCE_DIR}/ush/ioda/bufr2ioda::${PROJECT_SOURCE_DIR}/ush:${PROJECT_BINARY_DIR}/../lib/python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}:$ENV{PYTHONPATH}:${PROJECT_SOURCE_DIR}/../../../../ush/python")
```
Comment (guillaumevernieres) [inline test/soca/gw/CMakeLists.txt:81]: how is that change related to this PR @AndrewEichmann-NOAA ?

Comment (AndrewEichmann-NOAA) [reply test/soca/gw/CMakeLists.txt:81]: Because marine_letkf.py got moved to global-workflow - unless you want to link it into GDASApp

**Review by guillaumevernieres — APPROVED** [review]
:+1:

File: test/soca/gw/CMakeLists.txt (hunk 1 above)
Comment (guillaumevernieres) [reply test/soca/gw/CMakeLists.txt:81]: :man_facepalming:
